numpy/ma/core.py

```diff
@@ -219,7 +219,7 @@ def default_fill_value(obj):
         defval = default_filler['i']
     elif isinstance(obj, str):
         defval = default_filler['S']
-    elif isinstance(obj, unicode):
+    elif sys.version_info[0] < 3 and isinstance(obj, unicode):
```

Comment:
Might be better to import `unicode` and `bytes` from `numpy.compat` and use those instead, that way the two string types will be orthogonal for both Python 2 and Python 3.